### PR TITLE
Remove gcps

### PR DESCRIPTION
**Pull request by agrouaze**
address issue #73

### CODE REVIEW

**Review by oarcher — CHANGES_REQUESTED** [review]
From failing checks logs, `s1meta.geoloc.attrs['approx_transform']` is missing.

with `gcps` implementation, it was:
```python
        # approx transform, from all gcps (inaccurate)
        approx_transform = rasterio.transform.from_gcps(rio_gcps)
```

you will have to check the type of `rio_gcps`, so that the new function will be able to use the newly decoded variables.

**Review by oarcher — APPROVED** [review]

File: src/xsar/sentinel1_meta.py [hunk 1]
```diff
@@ -414,6 +308,25 @@ def geoloc(self):
...
+                    row=pt_geoloc.atrack.item(),
+                    col=pt_geoloc.xtrack.item()
+                )
+            rio_gcps = [
+                _to_rio_gcp(self._geoloc.sel(atrack=row, xtrack=col))
+                for row in  self._geoloc.atrack for col in self._geoloc.xtrack
+            ]
+            # approx transform, from all gcps (inaccurate)
+            approx_transform = rasterio.transform.from_gcps(rio_gcps)
+            self._geoloc.attrs['approx_transform'] = approx_transform
```
Comment (oarcher) [inline src/xsar/sentinel1_meta.py:329]: Since `self._geoloc.attrs['approx_transform']` is now defined, you can probably try to clean more code

**Review by oarcher — CHANGES_REQUESTED** [review]

File: src/xsar/sentinel1_meta.py (hunk 1 above)
Comment (oarcher) [reply src/xsar/sentinel1_meta.py:329]: Or perhaps all is already cleaned ? Do you think we are ready to merge @agrouaze ?

Comment (agrouaze) [reply src/xsar/sentinel1_meta.py:329]: Indeed I will define the `approx_transform`. The structure seems simple to replicate using `geoloc`:
image

**Comment by agrouaze** [comment]
notebook and doc related to TOPS are cleaned. ready to merge.

**Review by oarcher — APPROVED** [review]

**Pull request merged**